### PR TITLE
tso: pre-initialize the TSO metrics with the group label

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #7011.

### What is changed and how does it work?

```commit-message
Pre-initialize the TSO metrics with the group label.
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

image

image

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: pkg/tso/tso.go [hunk 1]
```diff
@@ -74,6 +76,65 @@ type timestampOracle struct {
 	lastSavedTime atomic.Value // stored as time.Time
 	suffix        int
 	dcLocation    string
+
+	// pre-initialized metrics
+	// timestampOracle event counter
+	syncEvent             prometheus.Counter
```
Comment (nolouch) [inline pkg/tso/tso.go:82]: How about putting it all into an independent structure? then like `t.metrics.syncEvent.Inc()`

File: pkg/tso/tso.go [hunk 2]
```diff
@@ -223,30 +284,30 @@ func (t *timestampOracle) resetUserTimestampInner(leadership *election.Leadershi
...
+		t.errResetSmallTSEvent.Inc()
 		if ignoreSmaller {
 			return nil
 		}
 		return errs.ErrResetUserTimestamp.FastGenByArgs("the specified ts is smaller than now")
 	}
 	// do not update if next logical time is less/before/equal than prev
 	if physicalDifference == 0 && logicalDifference <= 0 {
-		tsoCounter.WithLabelValues("err_reset_small_counter", t.dcLocation).Inc()
+		t.errResetSmallTSEvent.Inc()
```
Comment (rleungx) [inline pkg/tso/tso.go:295]: What is the different between line 287

File: pkg/tso/global_allocator.go [hunk 3]
```diff
@@ -281,10 +293,10 @@ func (gta *GlobalTSOAllocator) GenerateTSO(ctx context.Context, count uint32) (p
 			continue
 		}
 		if tsoutil.CompareTimestamp(currentGlobalTSO, &globalTSOResp) < 0 {
-			tsoCounter.WithLabelValues("global_tso_persist", gta.timestampOracle.dcLocation).Inc()
+			gta.timestampOracle.globalTSOPersistEvent.Inc()
 			// Update the Global TSO in memory
 			if err = gta.timestampOracle.resetUserTimestamp(ctx1, gta.member.GetLeadership(), tsoutil.GenerateTS(&globalTSOResp), true); err != nil {
-				tsoCounter.WithLabelValues("global_tso_persist_err", gta.timestampOracle.dcLocation).Inc()
+				gta.timestampOracle.globalTSOPersistErrEvent.Inc()
```
Comment (rleungx) [inline pkg/tso/global_allocator.go:299]: How about
```suggestion
				gta.timestampOracle.errGlobalTSOPersistEvent.Inc()
```

**Review by nolouch — APPROVED** [review]

**Pull request merged**